### PR TITLE
DOC: how to revert MultiIndex.to_flat_index

### PR DESCRIPTION
**Pull request by adamjstewart**
Took me way too long to figure this out. Hopefully this benefits someone else!

### CODE REVIEW

**Review by arw2019 — CHANGES_REQUESTED** [review]
Thanks for the PR!

Some formatting changes needed based on the CI run:
```
Error: /home/runner/work/pandas/pandas/pandas/core/indexes/multi.py:1779:GL07:pandas.MultiIndex.to_flat_index:Sections are in the wrong order. Correct order is: Returns, See Also, Notes, Examples
  warnings.warn(
Error: /home/runner/work/pandas/pandas/pandas/core/indexes/multi.py:1779:SA02:pandas.MultiIndex.to_flat_index:Missing period at end of description for See Also "MultiIndex.from_tuples" reference
```

**Review by MarcoGorelli — APPROVED** [review]
Nice, thanks @adamjstewart !

File: pandas/core/indexes/multi.py [hunk 1]
```diff
@@ -1787,6 +1787,10 @@ def to_flat_index(self):
         pd.Index
             Index with the MultiIndex data represented in Tuples.
 
+        See Also
+        --------
```
Comment (jreback) [inline pandas/core/indexes/multi.py:1791]: can you add an example int eh Examples section as well.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/multi.py [hunk 2]
```diff
@@ -1798,13 +1798,26 @@ def to_flat_index(self):
 
         Examples
         --------
+
+        To convert a MultiIndex to an Index:
+
         >>> index = pd.MultiIndex.from_product(
```
Comment (arw2019) [inline pandas/core/indexes/multi.py:1804]: might we want to name this something else for clarity, like `multiindex`

File: pandas/core/indexes/multi.py [hunk 3]
```diff
@@ -1798,13 +1798,26 @@ def to_flat_index(self):
...
         ...     [['foo', 'bar'], ['baz', 'qux']],
         ...     names=['a', 'b'])
-        >>> index.to_flat_index()
+        >>> index = index.to_flat_index()
+        >>> index
         Index([('foo', 'baz'), ('foo', 'qux'),
                ('bar', 'baz'), ('bar', 'qux')],
               dtype='object')
+
+        To convert an Index back to a MultiIndex:
```
Comment (arw2019) [inline pandas/core/indexes/multi.py:1813]: IMO this doesn't belong here - it's an example of `MultiIndex.from_tuples` constructor and not specifically to do with `to_flat_index`

Comment (adamjstewart) [reply pandas/core/indexes/multi.py:1813]: I'm getting some conflicting advice on this, so I'll let you decide amongst yourselves. I have no preference. 

My use case: I'm trying to merge two DataFrames, one uses a MultiIndex, the other uses an Index. The only way I found to get this to work is to collapse one MultiIndex and restore it later.

Comment (jreback) [reply pandas/core/indexes/multi.py:1813]: @adamjstewart yeah we want an example of using *to_flat_index*, actually we have one above, so I guess the reference that you added is ok, but yeah let's kill this example as this is not relevant for this function.

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @adamjstewart

**Pull request merged**